Bottom: 0efecf947edbbd6da7475ec629e919f5e60f5ee6
Top:    1fd799e2766ed8103fd3f02c3b34c147de6dc168
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-21 23:20:58 +0000

polygon.c: For some reason, the range computation in frac_circle appears to be wrong


---

diff --git a/src/polygon.c b/src/polygon.c
index 23400dd..4c7b1e0 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -405,8 +405,8 @@ frac_circle (PLINE * c, Coord X, Coord Y, Vector v, int fraction)
   e1 = (v[0] - X) * POLY_CIRC_RADIUS_ADJ;
   e2 = (v[1] - Y) * POLY_CIRC_RADIUS_ADJ;
 
-  /* NB: the caller adds the last vertex, hence the -1 */
-  range = POLY_CIRC_SEGS / fraction - 1;
+  /* XXX */ /* NB: the caller adds the last vertex, hence the -1 */
+  range = POLY_CIRC_SEGS / fraction;
   for (i = 0; i < range; i++)
     {
       /* rotate the vector */
